### PR TITLE
fix(search): Make event IDs case insensitive (lowercase)

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

File: src/sentry/db/models/manager.py [hunk 1]
```diff
@@ -271,7 +271,7 @@ def get_from_cache(self, **kwargs):
             if key != pk_name:
                 return self.get_from_cache(**{pk_name: retval})
 
-            if type(retval) != self.model:
+            if not isinstance(retval, self.model):
```
Comment (dcramer) [inline src/sentry/db/models/manager.py]: we concerned about this? auto formatter did it

Comment (dcramer) [reply src/sentry/db/models/manager.py]: (i realize its not quite the same, but i dont think it ultimately matters)

**Review by alex-hofsteede — APPROVED** [review]

**Review by mattrobenolt — APPROVED** [review]
This is fine if we're ok with making any non-lowercased events addressable. Unless we've been lowercasing already for over 90 days.

**Comment by dcramer** [comment]
@mattrobenolt thats what i dont have an answer to atm

**Comment by dcramer** [comment]
@alex-hofsteede do you have a way to query ... uppercase A-F characters in event_id in Snuba? :)

@mitsuhiko how long has normalization been live?

**Comment by mitsuhiko** [comment]
It has been live for 100% of projects from the 4th of Februrary. This means we probably have about two months of iOS events with non lowercased event id at least.

**Comment by dcramer** [comment]
Will defer to others on how we wanna tackle this. PR could be updated to look for both.

**Comment by dcramer** [comment]
So I'm guessing at this point we can only handle the lowercase path?

**Comment by BYK** [comment]
Closing due to staleness and conflicts.

**Pull request closed without merging**